### PR TITLE
CU-3cr19t9 - Change the contracts, transactions and blocks list pagination to the traditional style, so I can easily jump to the last page

### PR DESCRIPTION
**Pull request by Thacryba**

### CODE REVIEW

**Comment by lllwvlvwlll** [comment]
Task linked: [CU-3cr19t9 Change the contracts, transactions and blocks list pagination to the traditional style, so I can easily jump to the last page](https://app.clickup.com/t/3cr19t9)

File: src/pages/blocks/Blocks.tsx [hunk 1]
```diff
@@ -82,38 +88,27 @@ const Blocks: React.FC<MatchParams> = props => {
...
-    dispatch(fetchBlocks(nextPage))
-  }
   const history = useHistory()
   const { chain, network: networkParam } = useParams<MatchParams>()
   const { protocol, handleSetFilterData, network } = useFilterStateWithHistory(
     history,
     chain,
     networkParam,
   )
+  const [perPage, setPerPage] = useState<number>(60)
```
Comment (raulduartep) [inline src/pages/blocks/Blocks.tsx]: This default value is not always true, if you change the filter to something other than all and reload the page, the filter will start if the last change but the state not. So I recommend you start the create a useEffect. 
```suggestion
  const [perPage, setPerPage] = useState<number>()
  
  useEffect(() => {
     if (network !== "all") {
       setPage(15)
       return
     }
     
     setPage(60)
  }, [network])
```

File: src/pages/blocks/Blocks.tsx [hunk 2]
```diff
@@ -167,6 +162,10 @@ const Blocks: React.FC<MatchParams> = props => {
             },
           }}
           handleFilterUpdate={(option): void => {
+            model.events.goTo(1)
+            if ((option.value as Platform).network !== 'all') {
+              setPerPage(15)
+            }
```
Comment (raulduartep) [inline src/pages/blocks/Blocks.tsx]: If you did what I recommend you above you won't need this condicional

File: src/pages/blocks/Blocks.tsx [hunk 3]
```diff
@@ -207,14 +206,48 @@ const Blocks: React.FC<MatchParams> = props => {
...
+                    perPage,
+                  )}-${getVisibleResultsMax(
+                    state.currentPage,
+                    perPage,
+                    blockState.totalCount,
+                  )} of ${blockState.totalCount} results`
+                }
+              </Pagination.AdditionalDetails>
+            </Pagination>
+          </div>
```
Comment (raulduartep) [inline src/pages/blocks/Blocks.tsx]: what do you think about creating a separate component for this and reuse it on other pages ?

**Pull request merged**